internal/nginx/config/servers_template.go

```diff
@@ -44,6 +44,10 @@ server {
 		js_content httpmatches.redirect;
 		{{ end }}
 
+		{{- range $h := $l.ProxySetHeaders -}}
```

Comment:
since we only need to set headers when proxying, shall we move this block inside `{{- if $l.ProxyPass -}}` block?